lib/rules/no-else-return.js

```diff
@@ -86,7 +92,13 @@ module.exports = {
                     } else {
                         fixedSource = source;
                     }
-                    return fixer.replaceTextRange([elseToken.start, node.end], fixedSource);
+
+                    // Extend the replacement range to include the entire
+                    // function to avoid conflicting with no-useless-return.
+                    // https://github.com/eslint/eslint/issues/8026
+                    return new FixTracker(fixer, sourceCode)
+                        .retainEnclosingFunction(node)
+                        .replaceTextRange([elseToken.start, node.end], fixedSource);
```

Comment:
@mysticatea Since this fix might be a temporary workaround until #7928 is solved in a better way, we're trying to avoid making changes to core in this fix. So it adds `FixTracker` as an internal helper class, but it doesn't change the `ruleFixer` API because we're not sure how we'll want to change that in the future.